pandas/indexes/base.py

```diff
@@ -1949,8 +1949,6 @@ def difference(self, other):
         theDiff = sorted(set(self) - set(other))
         return Index(theDiff, name=result_name)
 
-    diff = deprecate('diff', difference)
-
```

Comment:
@jreback : I thought it would have been more appropriate in a separate PR, as this removal seemed a lot more straightforward than the `+` and `-` one (see my question <a href="https://github.com/pydata/pandas/issues/6581#issuecomment-233112384">here</a>).

What do you think?